certify-service/src/main/java/io/mosip/certify/advice/ExceptionHandlerAdvice.java

```diff
@@ -0,0 +1,228 @@
+import org.springframework.web.bind.MissingServletRequestParameterException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.ServletWebRequest;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.ConstraintViolationException;
+import java.io.IOException;
+import java.util.*;
+
+import static io.mosip.certify.core.constants.ErrorConstants.*;
+
+@Slf4j
+@ControllerAdvice
+public class ExceptionHandlerAdvice extends ResponseEntityExceptionHandler implements AccessDeniedHandler {
+
+    @Autowired
+    MessageSource messageSource;
+
+    @Override
+    protected ResponseEntity<Object> handleHttpMessageNotReadable(HttpMessageNotReadableException ex, HttpHeaders headers,
+                                                                  HttpStatusCode status, WebRequest request) {
+        return handleExceptions(ex, request);
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleHttpMediaTypeNotAcceptable(
+            HttpMediaTypeNotAcceptableException ex, HttpHeaders headers, HttpStatusCode status, WebRequest request) {
+        return handleExceptions(ex, request);
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleMissingServletRequestParameter(
+            MissingServletRequestParameterException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request) {
+        return handleExceptions(ex, request);
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatusCode status,
+            WebRequest request) {
+        return handleExceptions(ex, request);
+    }
+
+    protected ResponseEntity<Object> handleTypeMismatch(TypeMismatchException ex, HttpHeaders headers,
+                                                        HttpStatus status, WebRequest request) {
+        return handleExceptions(ex, request);
+    }
+
+    @ExceptionHandler(value = { Exception.class, RuntimeException.class, MissingRequestHeaderException.class })
+    public ResponseEntity handleExceptions(Exception ex, WebRequest request) {
+        log.error("Unhandled exception encountered in handler advice", ex);
+        String pathInfo = ((ServletWebRequest)request).getRequest().getPathInfo();
+
+        boolean isInternalAPI = pathInfo != null && (pathInfo.startsWith("/authorization") ||
+                pathInfo.startsWith("/client-mgmt/"));
+
+        if(!isInternalAPI && pathInfo.startsWith("/oidc/userinfo")) {
```

Comment:
I think we can remove code from lines 98-103, and also "handleExceptionWithHeader" will be no longer required.

Reply:
removed this check and exception with header method